traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/dns/ZoneManager.java

```diff
@@ -1009,7 +1015,20 @@ private static Zone fillDynamicZone(final LoadingCache<ZoneKey, Zone> dzc, final
 						continue;
 					}
 
-					records.add(r);
+					if (r.getType() == Type.NSEC && r.getName().toString().equals(staticZone.getOrigin().toString())) {
```

Comment:
We might need an `&& dsRoutingName != null` check here just for good measure, although it would likely never be null in the first place.